experimental/Schemes/IdealSheaves.jl

```diff
@@ -333,33 +333,28 @@ end
 For an ideal sheaf ``ℐ`` on an `AbsCoveredScheme` ``X`` return
 the subscheme ``Y ⊂ X`` given by the zero locus of ``ℐ``.
 """
-function subscheme(I::AbsIdealSheaf)
+function subscheme(I::AbsIdealSheaf; covering::Covering=default_covering(scheme(I)))
   X = space(I)
-  C = default_covering(X)
-  new_patches = [subscheme(U, I(U)) for U in basic_patches(C)]
+  C = covering
+  new_patches = IdDict{AbsAffineScheme, AbsAffineScheme}([U=>subscheme(U, I(U)) for U in basic_patches(C) if !isone(I(U))])
   new_gluings = IdDict{Tuple{AbsAffineScheme, AbsAffineScheme}, AbsGluing}()
   decomp_dict = IdDict{AbsAffineScheme, Vector{RingElem}}()
-  for (U, V) in keys(gluings(C))
-    i = C[U]
-    j = C[V]
-    Unew = new_patches[i]
-    Vnew = new_patches[j]
-    G = C[U, V]
-    #new_gluings[(Unew, Vnew)] = restrict(C[U, V], Unew, Vnew, check=false)
-    new_gluings[(Unew, Vnew)] = LazyGluing(Unew, Vnew, _compute_restriction,
+  for (U, Unew) in new_patches
+    for (V, Vnew) in new_patches
+      (U, V) in keys(gluings(C)) || continue # No gluing before, no gluing after.
+      old_glue = C[U, V]
+      new_gluings[(Unew, Vnew)] = LazyGluing(Unew, Vnew, _compute_restriction,
```

Comment:
it seems to me that the variable `_compute_restriction` is undefined?

Reply:
It is a function which is passed on as an argument. 

Maybe this just reads a little odd?

Reply:
No, it is the function name to be used in the lazy gluing. A function of this name is available in LazyGluing.jl.